### ISSUE TITLE
compile errors on Windows

### ISSUE THREAD

**Issue by syeg** [littlevgl/pc_simulator_sdl_eclipse#47]
Trying to compile PC Simulator on Windows 10 after following  instruction on https://docs.littlevgl.com/en/html/get-started/pc-simulator.html

BUT ...got compile error due to the fact that lv_drivers/display/fbdev.c has  include statements such as

#include <linux/fb.h>
#include <sys/mman.h>
#include <sys/ioctl.h>

Where do you download the include files?

Thx.

-------------
```
18:21:08 **** Incremental Build of configuration Debug for project pc_simulator ****
mingw32-make all 
'Building file: ../lv_drivers/display/fbdev.c'
'Invoking: Cross GCC Compiler'
gcc -I"D:\Eclipse_Cpp\pc_simulator_sdl_eclipse" -I"D:\Eclipse_Cpp\mingw-w64\i686-8.1.0-posix-dwarf-rt_v6-rev0\mingw32\bin" -O0 -g3 -Wall -c -fmessage-length=0 -Wall -W -MMD -MP -MF"lv_drivers/display/fbdev.d" -MT"lv_drivers/display/fbdev.o" -o "lv_drivers/display/fbdev.o" "../lv_drivers/display/fbdev.c"
../lv_drivers/display/fbdev.c:17:10: fatal error: linux/fb.h: No such file or directory
 #include <linux/fb.h>
          ^~~~~~~~~~~~
compilation terminated.
mingw32-make: *** [lv_drivers/display/subdir.mk:35: lv_drivers/display/fbdev.o] Error 1
"mingw32-make all" terminated with exit code 2. Build might be incomplete.
```

**Comment by embeddedt** [comment]
It looks like we are enabling `USE_FBDEV` for some reason. That is causing the compilation errors.

As a workaround, change [this](https://github.com/littlevgl/pc_simulator_sdl_eclipse/blob/5e1039be56ab933c19aa98447bbd119062ec1e5a/lv_drv_conf.h#L186) to `0` instead of `1`. That should fix the issue.

**Comment by syeg** [comment]
Thanks for the tip. I tried the fix nut encountered new errors:
16:23:42 **** Build of configuration Debug for project pc_simulator ****
mingw32-make all 
'Building target: pc_simulator'
'Invoking: Cross GCC Linker'
gcc -L"D:\Eclipse_Cpp\mingw-w64\i686-8.1.0-posix-dwarf-rt_v6-rev0\mingw32\lib\gcc\i686-w64-mingw32\lib" -o "pc_simulator"  ./lvgl/src/lv_themes/lv_theme.o ./lvgl/src/lv_themes/lv_theme_alien.o ./lvgl/src/lv_themes/lv_theme_default.o ./lvgl/src/lv_themes/lv_theme_material.o ./lvgl/src/lv_themes/lv_theme_mono.o ./lvgl/src/lv_themes/lv_theme_nemo.o ./lvgl/src/lv_themes/lv_theme_night.o ./lvgl/src/lv_themes/lv_theme_templ.o ./lvgl/src/lv_themes/lv_theme_zen.o  ./lvgl/src/lv_objx/lv_arc.o ./lvgl/src/lv_objx/lv_bar.o ./lvgl/src/lv_objx/lv_btn.o ./lvgl/src/lv_objx/lv_btnm.o ./lvgl/src/lv_objx/lv_calendar.o ./lvgl/src/lv_objx/lv_canvas.o ./lvgl/src/lv_objx/lv_cb.o ./lvgl/src/lv_objx/lv_chart.o ./lvgl/src/lv_objx/lv_cont.o ./lvgl/src/lv_objx/lv_ddlist.o ./lvgl/src/lv_objx/lv_gauge.o ./lvgl/src/lv_objx/lv_img.o ./lvgl/src/lv_objx/lv_imgbtn.o ./lvgl/src/lv_objx/lv_kb.o ./lvgl/src/lv_objx/lv_label.o ./lvgl/src/lv_objx/lv_led.o ./lvgl/src/lv_objx/lv_line.o ./lvgl/src/lv_objx/lv_list.o ./lvgl/src/lv_objx/lv_lmeter.o ./lvgl/src/lv_objx/lv_mbox.o ./lvgl/src/lv_objx/lv_objx_templ.o ./lvgl/src/lv_objx/lv_page.o ./lvgl/src/lv_objx/lv_preload.o ./lvgl/src/lv_objx/lv_roller.o ./lvgl/src/lv_objx/lv_slider.o ./lvgl/src/lv_objx/lv_spinbox.o ./lvgl/src/lv_objx/lv_sw.o ./lvgl/src/lv_objx/lv_ta.o ./lvgl/src/lv_objx/lv_table.o ./lvgl/src/lv_objx/lv_tabview.o ./lvgl/src/lv_objx/lv_tileview.o ./lvgl/src/lv_objx/lv_win.o  ./lvgl/src/lv_misc/lv_anim.o ./lvgl/src/lv_misc/lv_area.o ./lvgl/src/lv_misc/lv_async.o ./lvgl/src/lv_misc/lv_circ.o ./lvgl/src/lv_misc/lv_color.o ./lvgl/src/lv_misc/lv_fs.o ./lvgl/src/lv_misc/lv_gc.o ./lvgl/src/lv_misc/lv_ll.o ./lvgl/src/lv_misc/lv_log.o ./lvgl/src/lv_misc/lv_math.o ./lvgl/src/lv_misc/lv_mem.o ./lvgl/src/lv_misc/lv_task.o ./lvgl/src/lv_misc/lv_templ.o ./lvgl/src/lv_misc/lv_txt.o ./lvgl/src/lv_misc/lv_utils.o  ./lvgl/src/lv_hal/lv_hal_disp.o ./lvgl/src/lv_hal/lv_hal_indev.o ./lvgl/src/lv_hal/lv_hal_tick.o  ./lvgl/src/lv_font/lv_font.o ./lvgl/src/lv_font/lv_font_fmt_txt.o ./lvgl/src/lv_font/lv_font_roboto_12.o ./lvgl/src/lv_font/lv_font_roboto_16.o ./lvgl/src/lv_font/lv_font_roboto_22.o ./lvgl/src/lv_font/lv_font_roboto_28.o ./lvgl/src/lv_font/lv_font_unscii_8.o  ./lvgl/src/lv_draw/lv_draw.o ./lvgl/src/lv_draw/lv_draw_arc.o ./lvgl/src/lv_draw/lv_draw_basic.o ./lvgl/src/lv_draw/lv_draw_img.o ./lvgl/src/lv_draw/lv_draw_label.o ./lvgl/src/lv_draw/lv_draw_line.o ./lvgl/src/lv_draw/lv_draw_rect.o ./lvgl/src/lv_draw/lv_draw_triangle.o ./lvgl/src/lv_draw/lv_img_cache.o ./lvgl/src/lv_draw/lv_img_decoder.o  ./lvgl/src/lv_core/lv_disp.o ./lvgl/src/lv_core/lv_group.o ./lvgl/src/lv_core/lv_indev.o ./lvgl/src/lv_core/lv_obj.o ./lvgl/src/lv_core/lv_refr.o ./lvgl/src/lv_core/lv_style.o  ./lvgl/porting/lv_port_disp_template.o ./lvgl/porting/lv_port_fs_template.o ./lvgl/porting/lv_port_indev_template.o  ./lv_examples/lv_tutorial/9_responsive/lv_tutorial_responsive.o  ./lv_examples/lv_tutorial/8_animations/lv_tutorial_animations.o  ./lv_examples/lv_tutorial/7_fonts/arial_20.o ./lv_examples/lv_tutorial/7_fonts/lv_tutorial_fonts.o  ./lv_examples/lv_tutorial/6_images/flower_icon_alpha.o ./lv_examples/lv_tutorial/6_images/lv_tutorial_images.o ./lv_examples/lv_tutorial/6_images/red_flower.o ./lv_examples/lv_tutorial/6_images/red_rose_16.o  ./lv_examples/lv_tutorial/5_antialiasing/apple_icon_alpha.o ./lv_examples/lv_tutorial/5_antialiasing/apple_icon_chroma.o ./lv_examples/lv_tutorial/5_antialiasing/lv_tutorial_antialiasing.o  ./lv_examples/lv_tutorial/4_themes/lv_tutorial_themes.o  ./lv_examples/lv_tutorial/3_styles/lv_tutorial_styles.o  ./lv_examples/lv_tutorial/2_objects/lv_tutorial_objects.o  ./lv_examples/lv_tutorial/1_hello_world/lv_tutorial_hello_world.o  ./lv_examples/lv_tutorial/10_keyboard/lv_tutorial_keyboard.o  ./lv_examples/lv_tests/lv_test_theme/lv_test_theme_1.o ./lv_examples/lv_tests/lv_test_theme/lv_test_theme_2.o  ./lv_examples/lv_tests/lv_test_stress/lv_test_stress.o  ./lv_examples/lv_tests/lv_test_objx/lv_test_win/lv_test_win.o  ./lv_examples/lv_tests/lv_test_objx/lv_test_tileview/lv_test_tileview.o  ./lv_examples/lv_tests/lv_test_objx/lv_test_tabview/lv_test_tabview.o  ./lv_examples/lv_tests/lv_test_objx/lv_test_table/lv_test_table.o  ./lv_examples/lv_tests/lv_test_objx/lv_test_ta/lv_test_ta.o  ./lv_examples/lv_tests/lv_test_objx/lv_test_sw/lv_test_sw.o  ./lv_examples/lv_tests/lv_test_objx/lv_test_slider/lv_test_slider.o  ./lv_examples/lv_tests/lv_test_objx/lv_test_roller/lv_test_roller.o  ./lv_examples/lv_tests/lv_test_objx/lv_test_preload/lv_test_preload.o  ./lv_examples/lv_tests/lv_test_objx/lv_test_page/lv_test_page.o  ./lv_examples/lv_tests/lv_test_objx/lv_test_mbox/lv_test_mbox.o  ./lv_examples/lv_tests/lv_test_objx/lv_test_lmeter/lv_test_lmeter.o  ./lv_examples/lv_tests/lv_test_objx/lv_test_list/lv_test_list.o  ./lv_examples/lv_tests/lv_test_objx/lv_test_line/lv_test_line.o  ./lv_examples/lv_tests/lv_test_objx/lv_test_led/lv_test_led.o  ./lv_examples/lv_tests/lv_test_objx/lv_test_label/lv_test_label.o  ./lv_examples/lv_tests/lv_test_objx/lv_test_kb/lv_test_kb.o  ./lv_examples/lv_tests/lv_test_objx/lv_test_imgbtn/imgbtn_img_1.o ./lv_examples/lv_tests/lv_test_objx/lv_test_imgbtn/imgbtn_img_2.o ./lv_examples/lv_tests/lv_test_objx/lv_test_imgbtn/imgbtn_img_3.o ./lv_examples/lv_tests/lv_test_objx/lv_test_imgbtn/imgbtn_img_4.o ./lv_examples/lv_tests/lv_test_objx/lv_test_imgbtn/lv_test_imgbtn.o  ./lv_examples/lv_tests/lv_test_objx/lv_test_img/img_flower_icon.o ./lv_examples/lv_tests/lv_test_objx/lv_test_img/lv_test_img.o  ./lv_examples/lv_tests/lv_test_objx/lv_test_gauge/lv_test_gauge.o  ./lv_examples/lv_tests/lv_test_objx/lv_test_ddlist/lv_test_ddlist.o  ./lv_examples/lv_tests/lv_test_objx/lv_test_cont/lv_test_cont.o  ./lv_examples/lv_tests/lv_test_objx/lv_test_chart/lv_test_chart.o  ./lv_examples/lv_tests/lv_test_objx/lv_test_cb/lv_test_cb.o  ./lv_examples/lv_tests/lv_test_objx/lv_test_canvas/lv_test_canvas.o  ./lv_examples/lv_tests/lv_test_objx/lv_test_btnm/lv_test_btnm.o  ./lv_examples/lv_tests/lv_test_objx/lv_test_btn/lv_test_btn.o  ./lv_examples/lv_tests/lv_test_objx/lv_test_bar/lv_test_bar.o  ./lv_examples/lv_tests/lv_test_objx/lv_test_arc/lv_test_arc.o  ./lv_examples/lv_tests/lv_test_obj/lv_test_obj.o  ./lv_examples/lv_tests/lv_test_misc/lv_test_task.o  ./lv_examples/lv_tests/lv_test_group/lv_test_group.o  ./lv_examples/lv_apps/tpcal/tpcal.o  ./lv_examples/lv_apps/terminal/terminal.o  ./lv_examples/lv_apps/sysmon/sysmon.o  ./lv_examples/lv_apps/demo/demo.o ./lv_examples/lv_apps/demo/img_bubble_pattern.o  ./lv_examples/lv_apps/benchmark/benchmark.o ./lv_examples/lv_apps/benchmark/benchmark_bg.o  ./lv_drivers/indev/AD_touch.o ./lv_drivers/indev/FT5406EE8.o ./lv_drivers/indev/XPT2046.o ./lv_drivers/indev/evdev.o ./lv_drivers/indev/keyboard.o ./lv_drivers/indev/libinput.o ./lv_drivers/indev/mouse.o ./lv_drivers/indev/mousewheel.o  ./lv_drivers/display/R61581.o ./lv_drivers/display/SSD1963.o ./lv_drivers/display/ST7565.o ./lv_drivers/display/UC1610.o ./lv_drivers/display/fbdev.o ./lv_drivers/display/monitor.o  ./lv_drivers/win_drv.o  ./main.o ./mouse_cursor_icon.o   -lmingw32 -lSDL2main -lSDL2 -lm
./lv_drivers/display/monitor.o: In function `monitor_init':
D:\Eclipse_Cpp\pc_simulator_sdl_eclipse\Debug/../lv_drivers/display/monitor.c:109: undefined reference to `SDL_CreateThread'
./lv_drivers/display/monitor.o: In function `monitor_sdl_clean_up':
D:\Eclipse_Cpp\pc_simulator_sdl_eclipse\Debug/../lv_drivers/display/monitor.c:272: undefined reference to `SDL_DestroyTexture'
D:\Eclipse_Cpp\pc_simulator_sdl_eclipse\Debug/../lv_drivers/display/monitor.c:273: undefined reference to `SDL_DestroyRenderer'
D:\Eclipse_Cpp\pc_simulator_sdl_eclipse\Debug/../lv_drivers/display/monitor.c:274: undefined reference to `SDL_DestroyWindow'
D:\Eclipse_Cpp\pc_simulator_sdl_eclipse\Debug/../lv_drivers/display/monitor.c:283: undefined reference to `SDL_Quit'
./lv_drivers/display/monitor.o: In function `monitor_sdl_init':
D:\Eclipse_Cpp\pc_simulator_sdl_eclipse\Debug/../lv_drivers/display/monitor.c:289: undefined reference to `SDL_Init'
D:\Eclipse_Cpp\pc_simulator_sdl_eclipse\Debug/../lv_drivers/display/monitor.c:291: undefined reference to `SDL_SetEventFilter'
./lv_drivers/display/monitor.o: In function `monitor_sdl_refr_core':
D:\Eclipse_Cpp\pc_simulator_sdl_eclipse\Debug/../lv_drivers/display/monitor.c:326: undefined reference to `SDL_PollEvent'
D:\Eclipse_Cpp\pc_simulator_sdl_eclipse\Debug/../lv_drivers/display/monitor.c:357: undefined reference to `SDL_Delay'
./lv_drivers/display/monitor.o: In function `window_create':
D:\Eclipse_Cpp\pc_simulator_sdl_eclipse\Debug/../lv_drivers/display/monitor.c:363: undefined reference to `SDL_CreateWindow'
D:\Eclipse_Cpp\pc_simulator_sdl_eclipse\Debug/../lv_drivers/display/monitor.c:370: undefined reference to `SDL_CreateRenderer'
D:\Eclipse_Cpp\pc_simulator_sdl_eclipse\Debug/../lv_drivers/display/monitor.c:372: undefined reference to `SDL_CreateTexture'
D:\Eclipse_Cpp\pc_simulator_sdl_eclipse\Debug/../lv_drivers/display/monitor.c:374: undefined reference to `SDL_SetTextureBlendMode'
./lv_drivers/display/monitor.o: In function `window_update':
D:\Eclipse_Cpp\pc_simulator_sdl_eclipse\Debug/../lv_drivers/display/monitor.c:390: undefined reference to `SDL_UpdateTexture'
D:\Eclipse_Cpp\pc_simulator_sdl_eclipse\Debug/../lv_drivers/display/monitor.c:395: undefined reference to `SDL_RenderClear'
D:\Eclipse_Cpp\pc_simulator_sdl_eclipse\Debug/../lv_drivers/display/monitor.c:403: undefined reference to `SDL_RenderCopy'
D:\Eclipse_Cpp\pc_simulator_sdl_eclipse\Debug/../lv_drivers/display/monitor.c:404: undefined reference to `SDL_RenderPresent'
./main.o: In function `hal_init':
D:\Eclipse_Cpp\pc_simulator_sdl_eclipse\Debug/../main.c:174: undefined reference to `SDL_CreateThread'
./main.o: In function `tick_thread':
D:\Eclipse_Cpp\pc_simulator_sdl_eclipse\Debug/../main.c:191: undefined reference to `SDL_Delay'
collect2.exe: error: ld returned 1 exit status
mingw32-make: *** [makefile:91: pc_simulator] Error 1
"mingw32-make all" terminated with exit code 2. Build might be incomplete.

16:23:52 Build Failed. 20 errors, 0 warnings. (took 10s.8ms)

**Comment by embeddedt** [comment]
You must have changed something else, as changing that setting alone shouldn't affect the build.

If you are on Windows, I suggest using the [CodeBlocks project](https://github.com/littlevgl/pc_simulator_win_codeblocks) instead.

**Comment by kisvegabor** [comment]
Or on Windows, you can use the [Visual Studio](https://github.com/littlevgl/pc_simulator_sdl_visual_studio) project too. Similarly to the Code blocks project it also works out of the box without further configuration.

**Comment by 4lexChapman** [comment]
@syeg The undefined reference errors are likely due to not installing the SDL2 headers/libraries correctly. Did you follow the directions [here](https://docs.littlevgl.com/en/html/get-started/pc-simulator.html#install-sdl-2)?

**Comment by syeg** [comment]
Yes....it's been a while now...I did it for Windows (10)  and after the instillation, there were no SDL2 related libs for compile so I have to downloaded it from somewhere and copied over the SDL2 libs to move ahead. Then I hit this compile error.

Then I moved to use CodeBlocks project as suggested and it did not compile successfully.

Then I tried it with Visual Studio and still no able to pass initial compile.

By this time I have spent enough time on this so I decide not to forge ahead...would be nice to have it fixed for WIndows.

Also would it work for WSL (Windows subsystem for Linux)?

Thanks a lot for the replies.
